File: app-miyou/src/main/java/com/miyou/app/domain/dialogue/port/LlmPort.java
package com.miyou.app.domain.dialogue.port;

import com.miyou.app.domain.dialogue.model.CompletionRequest;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface LlmPort {
	Flux<String> streamCompletion(CompletionRequest request);

	Mono<String> complete(CompletionRequest request);
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/dialogue/port/LlmPort.kt
package com.miyou.app.domain.dialogue.port

import com.miyou.app.domain.dialogue.model.CompletionRequest
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface LlmPort {
    fun streamCompletion(request: CompletionRequest): Flux<String>

    fun complete(request: CompletionRequest): Mono<String>
}
